tools/wake/timeline.cpp

```diff
@@ -0,0 +1,242 @@
+/* Copyright 2022 SiFive, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You should have received a copy of LICENSE.Apache2 along with
+ * this software. If not, you may obtain a copy at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+#include <algorithm>
+#include <map>
+#include <fstream>
+
+#include "timeline.h"
+#include "runtime/database.h"
+#include "util/execpath.h"
+
+struct JobNode {
+    JobReflection &job;
+    long parent_id = -1;
+    std::vector<long> dependencies;
+
+    long long starttime;
+    long long endtime;
+
+    long long hypothetical_starttime = 0;
+    long long hypothetical_endtime = -1;
```

Comment:
nit: I'm personally a big fan of using int64_t instead of `long long` since its the same no matter what you're compiling.

not a nit: Also the distinction between hypothetical times and non-hypothetical times isn't clear here. A comment could explain this here for the reader.

Reply:
Done